src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py

```diff
@@ -24,6 +25,15 @@ def datetime_type(string):
     return datetime_type
 
 
+def get_decimal_type():
+    def decimal_type(string):
+        try:
+            return Decimal(string)
+        except ValueError:
+            raise ValueError("the value passed cannot be converted to decimal")
+    return decimal_type
```

Comment:
You shouldn't need the wrapper function `get_decimal_type`.
Just define `decimal_type` as is then in your param definition, pass the function `decimal_type` instead of doing `get_decimal_type ()`

Reply:
Same with `get_datetime_type` above.